### ISSUE TITLE
Pagination throw exception when dismissing the $mddialog Angularjs

### ISSUE THREAD

**Issue by hvhoat** [michaelbromley/angularUtils#310]
The pagination works well but when displaying a dialog and dismiss then browser is frozen and show exception. I found out the root cause is **$state.reload();** but I had to use this method to reload and get date from API to update.

Error in console (debug mode):
   ** pagination directive: the itemsPerPage id argument (id: __default) does not match a registered pagination-id.**

```
Code js:
var useFullScreen = ($mdMedia('sm') || $mdMedia('xs')) && vm.customFullscreen;
            $mdDialog.show({
                    templateUrl: 'app/dev-blog/dialog.composer.html',
                    controller: 'DialogController',
                    controllerAs: 'vm',
                    parent: angular.element(document.body),
                    targetEvent: ev,
                    clickOutsideToClose: true,
                    fullscreen: useFullScreen,
                    resolve: { //do some thing}

                })
                .then(function() {
                     $state.reload();
                }, function() {
                     $state.reload();
                });

Code html:
<div dir-paginate="article in vm.articleList | orderBy:'-created_at' | itemsPerPage: 9">
        <div  ng-show="article.willShowEvent">
            <div layout="row" layout-align="start center">
                <h2 class="md-display-1">{{ article.title }}</h2>
                <div flex></div>
            </div>
            <div class="md-subhead" am-time-ago="article.created_at | amUtc | amLocal"></div>
            <p class="md-body-1" ng-bind-html="article.content"></p>
        </div>
    </div>
    <div layout-align="center center" layout="row">
        <dir-pagination-controls on-page-change="vm.pageChangeHandler(newPageNumber)" template-url="app/dev-blog/pagination.tpl.html"></dir-pagination-controls>
    </div>
```

**Comment by michaelbromley** [comment]
That error comes from [here](https://github.com/michaelbromley/angularUtils/blob/master/src/directives/pagination/dirPagination.js#L478) and means that when the pagination filter is executed, the `dir-pagination-controls` directive seems to not exist for some reason.

This could occur, for example, if the controls are inside an ngIf which has not yet become `true`. If the code you posted above is verbatim, then of course this is not the case. Is there some indication in your code (the parts you did not post here) that the controls directive may not be created after a state reload?

**Comment by hvhoat** [comment]
I try set a pagination-id="value" but this issue is still existence when call $state.reload(); Maybe when $state.reload() the service of directive is not create and issue occurs. I try using another way to reload: window.location.reload() when dismissing $mddialog and issue does not occur. But this way is not good because the page is reloaded global.

**Comment by LeleDev** [comment]
I noted, since the last upgrade (from 0.10.1 to 0.11.0), that if I switch from a state (using ui-router) to the same state with different url params, i get a this error in a loop:
```
angular.js:13236 pagination directive: the itemsPerPage id argument (id: myFirstPaginationID) does not match a registered pagination-id.
```
No error instead when I load that state for the first time, or using the previous release (0.10.1)

**Comment by michaelbromley** [comment]
Hi, this should be fixed in v0.11.1. The previous release introduced a bug which I then had to revert.

Re-open if the new version is still broken.
